app/src/main/java/org/mozilla/fenix/home/intent/HomeDeepLinkIntentProcessor.kt

```diff
@@ -0,0 +1,150 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.home.intent
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Build
+import android.os.Build.VERSION.SDK_INT
+import android.provider.Settings
+import androidx.core.os.bundleOf
+import androidx.navigation.NavController
+import mozilla.components.concept.engine.EngineSession
+import mozilla.components.support.base.log.logger.Logger
+import org.mozilla.fenix.BrowserDirection
+import org.mozilla.fenix.BuildConfig
+import org.mozilla.fenix.GlobalDirections
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.browser.browsingmode.BrowsingMode
+import org.mozilla.fenix.ext.alreadyOnDestination
+import org.mozilla.fenix.settings.SupportUtils
+
+/**
+ * Deep links in the form of `fenix://host` open different parts of the app.
+ */
+
+class HomeDeepLinkIntentProcessor(
+    private val activity: HomeActivity,
+) : HomeIntentProcessor {
+    private val logger = Logger("DeepLinkIntentProcessor")
+
+    override fun process(intent: Intent, navController: NavController, out: Intent): Boolean {
+        val scheme = intent.scheme?.equals(BuildConfig.DEEP_LINK_SCHEME, ignoreCase = true) ?: return false
+        return if (scheme) {
+            intent.data?.let { handleDeepLink(it, navController) }
+            true
+        } else {
+            false
+        }
+    }
+
+    @Suppress("ComplexMethod")
+    private fun handleDeepLink(deepLink: Uri, navController: NavController) {
+        handleDeepLinkSideEffects(deepLink)
+
+        val globalDirections = when (deepLink.host) {
+            "home", "enable_private_browsing" -> GlobalDirections.Home
+            "urls_bookmarks" -> GlobalDirections.Bookmarks
+            "urls_history" -> GlobalDirections.History
+            "settings" -> GlobalDirections.Settings
+            "turn_on_sync" -> GlobalDirections.Sync
+            "settings_search_engine" -> GlobalDirections.SearchEngine
+            "settings_accessibility" -> GlobalDirections.Accessibility
+            "settings_delete_browsing_data" -> GlobalDirections.DeleteData
+            "settings_addon_manager" -> GlobalDirections.SettingsAddonManager
+            "settings_logins" -> GlobalDirections.SettingsLogins
+            "settings_tracking_protection" -> GlobalDirections.SettingsTrackingProtection
+            // We'd like to highlight views within the fragment
+            // https://github.com/mozilla-mobile/fenix/issues/11856
+            // The current version of UI has these features in more complex screens.
+            "settings_privacy" -> GlobalDirections.Settings
+            "settings_wallpapers" -> GlobalDirections.WallpaperSettings
+            "home_collections" -> GlobalDirections.Home
+
+            else -> return
+        }
+
+        if (!navController.alreadyOnDestination(globalDirections.destinationId)) {
```

Comment:
I'm not sure if this is the correct line, but when launching the wallpaper settings UI from a deeplink, the title is missing. Maybe it's because of `ExternalDeepLinkIntentProcessor`?

https://user-images.githubusercontent.com/1370580/152835843-8888f3f3-1942-474a-a53a-0550595252d1.mp4

Reply:
Oh, maybe this was intentional?

Reply:
You are right. I believe since this action (HomeActivity to WallPaperFragment) was global, the back toolbar isn't being generated by nav_graph

Reply:
It looks like the `WallpaperSettingsFragment` doesn't have the `showToolbar` as other similar fragments for example 

 https://github.com/mozilla-mobile/fenix/blob/b7b601eee1580bfb21a32e253e58b7eae016a31e/app/src/main/java/org/mozilla/fenix/addons/AddonsManagementFragment.kt#L78

Reply:
That address the issue :)